### PR TITLE
docs: product detail page example

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

closes: #130 

StackBlitz preview: https://stackblitz.com/github/shopware/frontends/tree/docs/product-detail-page-example/examples/product-detail-page?embed=1&file=README.md

TODO:
- [x] fill docs section with corresponding description

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Screenshots (if applicable)

image
### Additional context

there are also small fixes  found along the way for composables-next package.

### CODE REVIEW

File: examples/product-detail-page/src/components/ProductView.vue [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  >("product-description");
+
+  const { addToCart } = useAddToCart(product);
+  const { loadAssociations, isLoading, productAssociations } =
+    useProductAssociations(product, {
+      associationContext: "cross-selling",
+    });
+  const openRecommendedTab = () => {
+    activeTab.value = "product-cross-sell";
+    loadAssociations({
```
Comment (patzick) [inline examples/product-detail-page/src/components/ProductView.vue:30]: we're invoking it also when you switch between tabs, couldn't we load it only once when tab is opened for the first time?

File: examples/product-detail-page/src/components/ProductView.vue [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  } from "@shopware-pwa/helpers-next";
+  import {
+    useAddToCart,
+    useProductPrice,
+    usePrice,
+    useProductAssociations,
+  } from "@shopware-pwa/composables-next";
+  import ProductCard from "./ProductCard.vue";
+
+  const props = defineProps(["product"]);
```
Comment (patzick) [inline examples/product-detail-page/src/components/ProductView.vue:16]: let's use TS props definition here

File: examples/product-detail-page/src/components/ProductView.vue [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+          {{ regulationPrice ? "*" : "" }}
+        </div>
+        <button
+          @click="addToCartProxy()"
+          class="mt-4 w-2/3 relative inline-flex items-center justify-center p-0.5 overflow-hidden text-sm font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-cyan-500 to-blue-500 group-hover:from-cyan-500 group-hover:to-blue-500 hover:text-white dark:text-white focus:ring-4 focus:outline-none focus:ring-cyan-200 dark:focus:ring-cyan-800"
+        >
+          <span
+            class="w-full relative px-5 py-2.5 transition-all ease-in duration-75 bg-white dark:bg-gray-900 rounded-md group-hover:bg-opacity-0"
+          >
+            🛍 Add to cart
```
Comment (patzick) [inline examples/product-detail-page/src/components/ProductView.vue:121]: I like this icon. Maybe we'd add it to the demo as well? (:

File: packages/composables/src/useProductAssociations.ts [hunk 4]
```diff
@@ -53,16 +53,16 @@ export function useProductAssociations(
     searchParams: ShopwareSearchParams;
   }) => {
     isLoading.value = true;
-    const method = params.method || "get";
+    const method = params?.method || "get";
```
Comment (patzick) [inline packages/composables/src/useProductAssociations.ts:56]: Params are obligatory. Only `method` can be undefined. We need to trust more in TS :)

File: packages/composables/src/useProductAssociations.ts [hunk 5]
```diff
@@ -53,16 +53,16 @@ export function useProductAssociations(
...
     try {
       if (method && method === "get") {
         const response = await invokeGet(
           {
             address: `${getProductDetailsEndpoint(
-              product.value.id
-            )}/${association}${params.searchParams || ""}`,
+              product.value?.id,
```
Comment (patzick) [inline packages/composables/src/useProductAssociations.ts:62]: The whole function should not be invoked if the product does not exist. Maybe check it at the beginning and throw an error?
What's in response from API at this point?

**Review by patzick — CHANGES_REQUESTED** [review]
Looks good, few questions and suggestions though :)

File: examples/product-detail-page/src/components/ProductView.vue (hunk 1 above)
Comment (mkucmus) [reply examples/product-detail-page/src/components/ProductView.vue:30]: yes, it's being invoked every time. I can improve it for sharing good practices of course :)

File: packages/composables/src/useProductAssociations.ts (hunk 5 above)
Comment (mkucmus) [reply packages/composables/src/useProductAssociations.ts:62]: we will get 400 error due to invalid uuid. I will revert this line and throw an error on missing product object

File: examples/product-detail-page/src/components/ProductView.vue (hunk 3 above)
Comment (mkucmus) [reply examples/product-detail-page/src/components/ProductView.vue:132]: added ✅

File: examples/product-detail-page/src/components/ProductCard.vue [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    getSmallestThumbnailUrl,
+    getSrcSetForMedia,
+  } from "@shopware-pwa/helpers-next";
+  import {
+    useAddToCart,
+    useProductPrice,
+    usePrice,
+  } from "@shopware-pwa/composables-next";
+
+  const props = defineProps(["product"]);
```
Comment (patzick) [inline examples/product-detail-page/src/components/ProductCard.vue:13]: Somehow this comment disappeared https://github.com/shopware/frontends/pull/149#discussion_r1169624755
Let's use proper `props` definitions :)

Comment (patzick) [reply examples/product-detail-page/src/components/ProductCard.vue:13]: och, the comment is for another component, we just missed this one :)

File: packages/composables/src/useProductAssociations.ts [hunk 7]
```diff
@@ -59,10 +62,10 @@ export function useProductAssociations(
         const response = await invokeGet(
           {
             address: `${getProductDetailsEndpoint(
-              product.value.id
+              product.value.id,
```
Comment (patzick) [inline packages/composables/src/useProductAssociations.ts:65]: is the TS okay with that? `getProductDetailsEndpoint` should take 1 param, and we're passing second `undefined` param here by accident

**Review by patzick — CHANGES_REQUESTED** [review]
just one more prop to fix :)

**Review by patzick — APPROVED** [review]
Great stuff!

**Pull request merged**